### PR TITLE
Add regex option to readTimeSeries

### PR DESCRIPTION
**Pull request by HomoCodens**
Would it be better to have a separate pattern param and ignore
series if regex == TRUE?

Relates to #27

### CODE REVIEW

File: R/readTimeseries.R [hunk 1]
```diff
@@ -25,7 +28,25 @@ readTimeSeries <- function(series, con,
                            schema = "timeseries",
                            env = NULL,
                            pkg_for_irreg = "xts",
-                           chunksize = 10000){
+                           chunksize = 10000,
+                           regex = FALSE){
+  
+  if(regex) {
+    if(length(series) > 1) {
+      stop("Only supports a single expression in series!")
```
Comment (mbannert) [inline R/readTimeseries.R:36]: This ain't true. Fuzzy searches can give a vector of keys back. Doesn't it make sense to return ALL series whose keys match?

File: R/readTimeseries.R [hunk 2]
```diff
@@ -25,7 +28,25 @@ readTimeSeries <- function(series, con,
...
+    }
+    
+    pattern <- series
+    
+    match_query <- sprintf("SELECT ts_key FROM %s.timeseries_main WHERE ts_key ~ '%s'",
```
Comment (mbannert) [inline R/readTimeseries.R:41]: Did you try do run this in SQL? You should definitely get more than one key back.

**Review by mbannert — CHANGES_REQUESTED** [review]
please have a look at my comments. logical regex param fine for though

File: R/readTimeseries.R (hunk 1 above)
Comment (HomoCodens) [reply R/readTimeseries.R:36]: What I mean is that only a single string should be supplied in series if `regex == TRUE`. i.e.
```
readTimeSeries(c("ch\\.kof.*", "us\\.datastream.*"), con, regex = TRUE)
```
would be illegal.
That's an arbitrary decision though, we could also do `ts_key ~ '(ch\.kof.*|us\.datastream.*)'` in that case.

**Comment by mbannert** [comment]
Can we merge this into test first in order to build and test from test? 
All of your changes look reasonable to me, but breaking timeseriesdb would have quite some consequences so we should proceed with great caution here.

**Comment by HomoCodens** [comment]
Sure, changed base to dev.

**Pull request merged**